### PR TITLE
Add com.github.rogercrocker.badabib

### PR DESCRIPTION
**Pull request by RogerCrocker**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
 
  I think "--filesystem=home" might not be necessary, but I don't know how to avoid it. (Should be fixed in the latest version.)

- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.github.rogercrocker.badabib.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
+{
+    "app-id" : "com.github.rogercrocker.badabib",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "40",
```
Comment (hfiguiere) [inline com.github.rogercrocker.badabib.json:4]:
```suggestion
    "runtime-version" : "41",
```

41 is out now

**Comment by hfiguiere** [comment]
bot, build com.github.rogercrocker.badabib

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.github.rogercrocker.badabib.json (hunk 1 above)
Comment (RogerCrocker) [reply com.github.rogercrocker.badabib.json:4]: Thanks, I missed that.

**Comment by RogerCrocker** [comment]
Sorry, I don't know how to fix the build failure. What does it mean that the "validation" of the screenshots failed?

**Comment by hfiguiere** [comment]
the screenshots listed in the appdata file can't be downloaded since the URL do NOT point to the actual image but to the github page.

**Comment by barthalion** [comment]
Yup, should be replaced with `https://raw.githubusercontent.com/RogerCrocker/BadaBib/master/data/screenshots/editor.png` (and so on for all the others).

**Comment by nedrichards** [comment]
If you want to drop filesystem permissions for a GTK app you need to use GTKFileChooserNative rather than GTKFileChooser to open files https://docs.gtk.org/gtk3/class.FileChooserNative.html in the basic case it should be a 'add native to the end of your call' but there are some cases that it doesn't support which may be important to your app so take a look and experiment.

**Comment by RogerCrocker** [comment]
OK, thanks a lot, that's very helpful. I'll look into these issues later today.

**Comment by RogerCrocker** [comment]
OK, I corrected the links and replaced GTKFileChooser with GTKFileChooserNative. Can you try again?

**Comment by hfiguiere** [comment]
bot, build com.github.rogercrocker.badabib

**Comment by RogerCrocker** [comment]
Great, thanks everyone!

**Pull request closed without merging**